mods/cnc/maps/gdi05b/gdi05b.lua

```diff
@@ -0,0 +1,197 @@
+}
+
+AtkRoute1 = { waypoint4.Location, waypoint5.Location, waypoint6.Location, waypoint7.Location, waypoint8.Location }
+AtkRoute2 = { waypoint0.Location, waypoint1.Location, waypoint2.Location, waypoint3.Location }
+
+AutoCreateTeams = 
+{
+	{ { ['e1'] = 1,  ['e3'] = 3 }, AtkRoute2 },
+	{ { ['e1'] = 3,  ['e3'] = 1 }, AtkRoute2 },
+	{ { ['e3'] = 4 }             , AtkRoute1 },
+	{ { ['e1'] = 4 }             , AtkRoute1 },
+	{ { ['bggy'] = 1 }           , AtkRoute1 },
+	{ { ['bggy'] = 1 }           , AtkRoute2 },
+	{ { ['ltnk'] = 1 }           , AtkRoute1 },
+	{ { ['ltnk'] = 1 }           , AtkRoute2 }
+}
+
+RepairThreshold = 0.6
+
+Atk1Delay = DateTime.Seconds(40)
+Atk2Delay = DateTime.Seconds(60)
+Atk3Delay = DateTime.Seconds(70)
+Atk4Delay = DateTime.Seconds(90)
+AutoAtkStartDelay = DateTime.Seconds(115)
+AutoAtkMinDelay = DateTime.Seconds(45)
+AutoAtkMaxDelay = DateTime.Seconds(90)
+
+Atk5CellTriggers =
+{
+	CPos.New(17,55), CPos.New(16,55), CPos.New(15,55), CPos.New(50,54), CPos.New(49,54),
+	CPos.New(48,54), CPos.New(16,54), CPos.New(15,54), CPos.New(14,54), CPos.New(50,53),
+	CPos.New(49,53), CPos.New(48,53), CPos.New(50,52), CPos.New(49,52)
+}
+
+GdiBase = { GdiNuke1, GdiProc1, GdiWeap1, GdiNuke2, GdiPyle1, GdiSilo1, GdiSilo2, GdiHarv }
+GdiUnits = { "e2", "e2", "e2", "e2", "e1", "e1", "e1", "e1", "mtnk", "mtnk", "jeep", "jeep", "apc" }
+NodSams = { Sam1, Sam2, Sam3, Sam4 }
+
+AllToHunt = function()
+	local list = nod.GetGroundAttackers()
+	Utils.Do(list, function(unit)
+		unit.Hunt()
+	end)
+end
+
+MoveThenHunt = function(actors, path)
+	Utils.Do(actors, function(actor)
+		actor.Patrol(path, false)
+		IdleHunt(actor)
+	end)
+end
+
+AutoCreateTeam = function()
+	local team = Utils.Random(AutoCreateTeams)
+	local MyActors = nod.GetActorsByType(team[1])
+	
+	MoveThenHunt(MyActors, team[2])
+
+	Trigger.AfterDelay(Utils.RandomInteger(AutoAtkMinDelay, AutoAtkMaxDelay), AutoCreateTeam)
+end
+
+DiscoverGdiBase = function(actor, discoverer)
+	if baseDiscovered or not discoverer == gdi then
+		return
+	end
+
+	Utils.Do(GdiBase, function(actor)
+		actor.Owner = gdi
+	end)
+	GdiHarv.FindResources()
+
+	baseDiscovered = true
+
+	gdiObjective3 = gdi.AddPrimaryObjective("Eliminate all Nod forces in the area")
+	gdi.MarkCompletedObjective(gdiObjective1)
+end
+
+Atk1TriggerFunction = function()
+    local MyActors = nod.GetActorsByType({ ['e1'] = 2,  ['e3'] = 3 })  
+	MoveThenHunt(MyActors, AtkRoute1)
+end
+
+Atk2TriggerFunction = function()
+	local MyActors = nod.GetActorsByType({ ['e1'] = 3,  ['e3'] = 3 })
+	MoveThenHunt(MyActors, AtkRoute2)
+end
+
+Atk3TriggerFunction = function()
+	local MyActors = nod.GetActorsByType({ ['bggy'] = 1 })
+	MoveThenHunt(MyActors, AtkRoute1)
+end
+
+Atk4TriggerFunction = function()
+	local MyActors = nod.GetActorsByType({ ['bggy'] = 2 })
+	MoveThenHunt(MyActors, AtkRoute2)
+end
+
+Atk5TriggerFunction = function()
+	local MyActors = nod.GetActorsByType({ ['ltnk'] = 1 })
```

Comment:
My idea was that all those should translate to something like `nod.GetActorsByType("1tnk").Take(1)` instead.